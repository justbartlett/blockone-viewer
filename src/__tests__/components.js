import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Blocks from '../components/blocks';
import Block from '../components/block';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Pacman from '../components/pacman';
import ErrorMessage from '../components/errorMessage';

Enzyme.configure({ adapter: new Adapter() });

describe('the Blocks component', () => {
  it('should render a Block component for each block', () => {
    const blocks = [
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436ee',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 20
      },
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e1',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 17
      },
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e2',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 15
      }
    ];
    const wrapper = shallow(<Blocks blocks={blocks} />);
    expect(wrapper.find('Block').length).toEqual(3);
  });
  it('should match the last snapshot without blocks', () => {
    const wrapper = shallow(<Blocks blocks={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should match the last snapshot with blocks', () => {
    const blocks = [
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436ee',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 20
      },
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e1',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 17
      },
      {
        id: '06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e2',
        timestamp: '2020-03-28T16:34:54.500',
        actions: 15
      }
    ];
    const wrapper = shallow(<Blocks blocks={blocks} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should match a snapshot with pacman if loading', () => {
    const wrapper = shallow(<Blocks isLoading="true" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('the Block component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Block
        block={{
          id: 1,
          timestamp: '1',
          transactions: [
            {
              trx: [{ id: 1 }]
            }
          ]
        }}
        blockIndex="1"
      />
    );
  });
  it('should match the last snapshot with a block', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('the Pacman component', () => {
  it('should match the last Pacman snapshot', () => {
    const wrapper = shallow(<Pacman />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('the ErrorMessage component', () => {
  it('should match the last ErrorMessage snapshot', () => {
    const wrapper = shallow(<ErrorMessage message="oh no, error" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
