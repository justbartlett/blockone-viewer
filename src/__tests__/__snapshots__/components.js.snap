// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Block component should match the last snapshot with a block 1`] = `
<div
  className="block"
  style={
    Object {
      "animationDelay": "0.05s",
    }
  }
>
  <div
    className="block-wrapper"
  >
    <div
      className="block-item"
      key="0"
    >
      <div
        className="block-item-label"
      >
        id
      </div>
      <div
        className="block-item-value"
      >
        1
      </div>
    </div>
    <div
      className="block-item"
      key="1"
    >
      <div
        className="block-item-label"
      >
        timestamp
      </div>
      <div
        className="block-item-value"
      >
        1
      </div>
    </div>
    <div
      className="block-item"
      key="2"
    >
      <div
        className="block-item-label"
      >
        actions
      </div>
      <div
        className="block-item-value"
      >
        1
      </div>
    </div>
    <div
      className="block-content"
      style={
        Object {
          "maxHeight": "0px",
        }
      }
    >
      <div
        className="block-item"
        key="0"
      >
        <div
          className="block-item-label"
        >
          block #
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="1"
      >
        <div
          className="block-item-label"
        >
          producer
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="2"
      >
        <div
          className="block-item-label"
        >
          confirmed
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="3"
      >
        <div
          className="block-item-label"
        >
          previous
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="4"
      >
        <div
          className="block-item-label"
        >
          transaction merkle root
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="5"
      >
        <div
          className="block-item-label"
        >
          action merkle root
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="6"
      >
        <div
          className="block-item-label"
        >
          schedule version
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="7"
      >
        <div
          className="block-item-label"
        >
          new producers
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-item"
        key="8"
      >
        <div
          className="block-item-label"
        >
          producer signature
        </div>
        <div
          className="block-item-value"
        />
      </div>
      <div
        className="block-transactions"
      >
        <div
          className="block-transaction-row"
        >
          <div
            className="block-transaction-col-large label"
          >
            Transactions - ID
          </div>
          <div
            className="block-transaction-col label"
          >
            Status
          </div>
          <div
            className="block-transaction-col label"
          >
            CPU Usage
          </div>
          <div
            className="block-transaction-col label"
          >
            NET Usage
          </div>
          <div
            className="block-transaction-col label"
          >
            Actions
          </div>
        </div>
        <div
          className="block-transaction-row"
          key="0"
        >
          <div
            className="block-transaction-col-large"
          />
          <div
            className="block-transaction-col"
          />
          <div
            className="block-transaction-col"
          />
          <div
            className="block-transaction-col"
          />
          <div
            className="block-transaction-col"
          />
        </div>
      </div>
    </div>
    <Ripple
      style={
        Object {
          "display": "block",
          "width": "100%",
        }
      }
    >
      <button
        className="block-button"
        onClick={[Function]}
      >
        <div
          className="text disabled "
        >
          Less
        </div>
        <div
          className="text active "
        >
          Details
        </div>
      </button>
    </Ripple>
  </div>
</div>
`;

exports[`the Blocks component should match the last snapshot with blocks 1`] = `
<div
  className="blocks"
>
  <Block
    block={
      Object {
        "actions": 20,
        "id": "06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436ee",
        "timestamp": "2020-03-28T16:34:54.500",
      }
    }
    blockIndex={0}
    key="0"
  />
  <Block
    block={
      Object {
        "actions": 17,
        "id": "06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e1",
        "timestamp": "2020-03-28T16:34:54.500",
      }
    }
    blockIndex={1}
    key="1"
  />
  <Block
    block={
      Object {
        "actions": 15,
        "id": "06b570581dbc8cc398ac80b2dc98683701f1d7ef32dbabea4bfff7402ef436e2",
        "timestamp": "2020-03-28T16:34:54.500",
      }
    }
    blockIndex={2}
    key="2"
  />
</div>
`;

exports[`the Blocks component should match the last snapshot without blocks 1`] = `
<div
  className="blocks"
/>
`;

exports[`the ErrorMessage component should match the last ErrorMessage snapshot 1`] = `
<div
  className="error-message"
>
  oh no, error
</div>
`;

exports[`the Pacman component should match the last Pacman snapshot 1`] = `
<div
  className="pacman"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
